app/file_list/file_list.controller.js

```diff
@@ -23,7 +23,7 @@ controller('FileListController', ['$scope', '$routeSegment', 'SelectedFiles', 'T
 
   vm.select_all = () => {
     if (!vm.all_selected) {
-      vm.selected = SelectedFiles.selected.map(file => file.id);
+      vm.selected = Facet.filter($filter('filter')(SelectedFiles.selected, {'type': 'file'})).map(file => file.id);
```

Comment:
I wonder if this might be clearer by using native JS `.filter` instead of `$filter`?

```js
vm.selected = Facet.filter(SelectedFiles.selected.filter(file => file.type === 'file')).map(file => file.id);
```